zerver/lib/actions.py

```diff
@@ -5946,13 +5959,16 @@ def do_update_message(
         event["prior_mention_user_ids"] = list(prior_mention_user_ids)
         event["mention_user_ids"] = list(mention_user_ids)
         event["presence_idle_user_ids"] = filter_presence_idle_user_ids(info["active_user_ids"])
-        if target_message.mentions_wildcard:
+        if rendering_result.mentions_wildcard:
             event["wildcard_mention_user_ids"] = list(info["wildcard_mention_user_ids"])
         else:
             event["wildcard_mention_user_ids"] = []
 
         do_update_mobile_push_notification(
-            target_message, prior_mention_user_ids, info["stream_push_user_ids"]
+            target_message,
+            rendering_result.mentions_user_ids,
+            prior_mention_user_ids,
+            info["stream_push_user_ids"],
```

Comment:
This interface might be cleaner to just do a prep commit that refactors `do_update_mobile_push_notification` to accept `(target_message, prior_mention_user_ids, mentions_user_ids, stream_push_user_ids)` (I think that other parameter order seems clearer, as usually one passes old value before new) and then this line can just replace `target_message.mentions_user_ids` with `rendering_result.mentions_user_ids`.